cedar-policy-core/src/ast/policy_set.rs

```diff
@@ -264,6 +264,125 @@ impl PolicySet {
         Ok(())
     }
 
+    /// Helper function for merge_policyset to check if the `PolicyID` pid
+    /// appears in this `PolicySet`'s links or templates.
+    fn policy_id_is_bound(&self, pid: &PolicyID) -> bool {
+        match self.templates.get(pid) {
+            Some(_) => true,
+            None => match self.links.get(pid) {
+                Some(_) => true,
+                None => false,
+            },
+        }
+    }
+
+    /// Merges this `PolicySet` with another `PolicySet`.
+    /// This `PolicySet` is modified while the other `PolicySet`
+    /// remains unchanged.
+    ///
+    /// The flag `rename_duplicates` controls the expected behavior
+    /// when a `PolicyID` in this and the other `PolicySet` conflict.
+    ///
+    /// When `rename_duplicates` is false, conflicting `PolicyID`s result
+    /// in a occupied `PolicySetError`.
+    ///
+    /// Otherwise, when `rename_duplicates` is true, conflicting `PolicyID`s from
+    /// the other `PolicySet` are automatically renamed to avoid conflict.
+    /// This renaming is returned as a Hashmap from the old `PolicyID` to the
+    /// renamed `PolicyID`.
+    pub fn merge_policyset(
+        &mut self,
+        other: &PolicySet,
+        rename_duplicates: bool,
+    ) -> Result<HashMap<PolicyID, PolicyID>, PolicySetError> {
+        // Check for conflicting policy ids. If there is a conflict either
+        // Throw an error or consturct a renaming (if rename_duplicates is true)
```

Comment:
Typo here. You should also have "throw" (not capitalized) and can add backticks to `rename_duplicates` like before.